### PR TITLE
Test to close scala/bug#5075.

### PR DESCRIPTION
**Pull request by milessabin**
Fixed, most likely by the SI-2712 fix.

### CODE REVIEW

File: test/files/pos/t5075.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+object Test {
+    trait TypeWithHigherKindParam[D[_]]
+    implicit def function[D[_]](t: TypeWithHigherKindParam[D]) = 1
+    val param: TypeWithHigherKindParam[PartialApply1Of2[Tuple2, Int]#Apply] = null
+
+    function[PartialApply1Of2[Tuple2, Int]#Apply](param): Int //Compiles
+    function(param) //Does not compile
+    param: Int //Does not compile
```
Comment (Blaisorblade) [inline test/files/pos/t5075.scala]: Cool that it compiles now! Suspect those comments are stale 😉

File: test/files/pos/t5075.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
+//Copied from Scalaz - http://scalaz.googlecode.com/svn/continuous/latest/browse.sxr/scalaz/PartialApplys.scala.html
+trait PartialApply1Of2[T[_, _], A] {
+  type Apply[B] = T[A, B]
+
+  type Flip[B] = T[B, A]
```
Comment (S11001001) [inline test/files/pos/t5075.scala]: Was `Flip` to be used for something?

Comment (Blaisorblade) [reply test/files/pos/t5075.scala]: Guess I just forgot to minimize it away in 2011. But it's useful for your comment https://github.com/scala/bug/issues/5075#issuecomment-370148938.

Comment (S11001001) [reply test/files/pos/t5075.scala]: @Blaisorblade I suspect that a modern minimization of 5075 would use `Flip` instead, no?

Comment (Blaisorblade) [reply test/files/pos/t5075.scala]: Of course, I already upvoted your comment! (Nitpick: I'd add testcases with `Flip` in addition.)

**Review by adriaanm — CHANGES_REQUESTED** [review]
Echoing requested changes by other reviewers

**Comment by adriaanm** [comment]
I updated the test file.

**Review by adriaanm — APPROVED** [review]

**Pull request merged**